.github/workflows/build.yaml

```diff
@@ -0,0 +1,103 @@
+name: "Build & Test"
+
+on:
+  push:
+    branches:
+      - main
+      - "release/*"
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      override-leap-dev:
+        description: Override leap-dev target
+        type: string
+      override-leap-dev-prerelease:
+        type: choice
+        description: Override leap-dev prelease
+        options:
+        - default
+        - true
+        - false
+      override-cdt:
+        description: 'Override cdt target'
+        type: string
+      override-cdt-prerelease:
+        type: choice
+        description: Override cdt prelease
+        options:
+        - default
+        - true
+        - false
+
+defaults:
+  run:
+    shell: bash
+
+jobs:
+  build-test:
+    name: Build & Test
+    runs-on: ubuntu-20.04
+    steps:
+      - name: Setup leap-dev & cdt versions
+        id: versions
+        env:
+          GH_TOKEN: ${{secrets.GITHUB_TOKEN}}
+        run: |
+          DEFAULTS_JSON=$(curl -s -f $(gh api https://api.github.com/repos/${{github.repository}}/contents/.cicd/defaults.json?ref=${{github.sha}} --jq .download_url))
+          echo leap-dev-target=$(echo "$DEFAULTS_JSON" | jq -r '."leap-dev".target') >> $GITHUB_OUTPUT
```

Comment:
I like that you have `curl` using the `f` and `s` flags here, but I recommend adding the `S` flag to make debugging failures easier, and the `L` flag for resiliency. Failures will go to `STDERR`, not `STDOUT`.

[![curl flags](https://user-images.githubusercontent.com/34947245/199630908-3fefe16b-b281-4a49-b9d4-ca4ded9a2d95.png)](https://explainshell.com/explain?cmd=curl+-fsSL+example.com)

Reply:
Resolved.